flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+    utils.url = "github:gytis-ivaskevicius/flake-utils-plus";
+  };
+
+  outputs = inputs@{ self, utils, ... }:
+    utils.lib.mkFlake rec {
+      inherit self inputs;
+
+      supportedSystems = [
+        "aarch64-linux"
+        "aarch64-darwin"
+        "i686-linux"
+        "x86_64-darwin"
+        "x86_64-linux"
+      ];
+
+      outputsBuilder = channels: with channels; {
+        packages = with nixpkgs; { 
+          inherit (nixpkgs) package-from-overlays;
+
+          nixpacks = rustPlatform.buildRustPackage rec {
+            pname = "nixpacks";
+            version = "v0.0.20";
+            doCheck = true;
+            src = ./.;
+            nativeBuildInputs = [ clippy ];
+            # skip `cargo test` due tests FHS dependency
+            checkPhase = ''
+              cargo check
+              cargo clippy
+            '';
+            cargoLock = {
+              lockFile = ./Cargo.lock;
+            };
+            meta = with nixpkgs.lib; {
+              description = "App source + Nix packages + Docker = Image";
+              homepage = "https://github.com/railwayapp/nixpacks";
+              license = licenses.mit;
+              maintainers = [ maintainers.zoedsoupe ];
+            };
+          };
+        };
+
+        devShell = nixpkgs.mkShell {
+          name = "nixpacks";
+
+          buildInputs = with nixpkgs; [
+            rustc cargo rustfmt clippy docker
```

Comment:
It would also be good to have things like rustfix and other rust tools.

Reply:
of course! I'm kinda new to rust ecosystem so could you list some useful tools would you like to be in dev env?

Reply:
I added a "complete" toolchain in dev env, `rls` and `rustfix` included 😃